Documented GFX shards

ShahFragcolor opened the pull request:
- documented GFX shards GFX.Feature
GFX.BuiltinFeature
GFX.Drawable
GFX.Draw
GFX.DrawQueue
GFX.ClearQueue
GFX.QueueDrawables
GFX.DrawablePass
GFX.EffectPass
GFX.CopyPass
GFX.View
GFX.RenderInto
GFX.ViewProjectionMatrix
GFX.ViewMatrix
GFX.ViewRange
GFX.Render
GFX.BuiltinMesh
GFX.Mesh
GFX.glTF
GLTF.PackGLB
GFX.Material
GFX.UIPass
GFX.WindowPosition
GFX.WindowSize
GFX.MoveWindow
GFX.ResizeWindow


<!--
Before submitting your PR, please review the following checklist:

- [ ] **DO NOT** submit a PR without having discussed the change first in a related issue. If none exist, create one first.
- [ ] **CONSIDER** adding a unit test if your PR resolves an issue.
- [ ] **DO** keep pull requests small so they can be easily reviewed.
- [ ] **DO** make sure all unit tests pass.
- [ ] **DO** make sure not to introduce any new compiler warnings.
- [ ] **AVOID** breaking the continuous integration build.
- [ ] **AVOID** making significant changes to the overall architecture.
-->

ShahFragcolor committed:
Documented GFX shards

- documented GFX shards
GFX.Feature
GFX.BuiltinFeature
GFX.Drawable
GFX.Draw
GFX.DrawQueue
GFX.ClearQueue
GFX.QueueDrawables
GFX.DrawablePass
GFX.EffectPass
GFX.CopyPass
GFX.View
GFX.RenderInto
GFX.ViewProjectionMatrix
GFX.ViewMatrix
GFX.ViewRange
GFX.Render
GFX.BuiltinMesh
GFX.Mesh
GFX.glTF
GLTF.PackGLB
GFX.Material
GFX.UIPass
GFX.WindowPosition
GFX.WindowSize
GFX.MoveWindow
GFX.ResizeWindow

guusw commented on shards/modules/gfx/drawable.cpp:
```diff
@@ -290,7 +300,10 @@ struct GetQueueDrawablesShard {
   static SHTypesInfo inputTypes() { return ShardsTypes::DrawQueue; }
   static SHTypesInfo outputTypes() { return OutputSeqType; }
-  static SHOptionalString help() { return SHCCSTR("Retrieves the individual drawables in a draw queue"); }
+  static SHOptionalString help() { return SHCCSTR("Converts the input drawable queue object to a sequence of drawable objects."); }
```
Maybe 'convert' is not the best word, more likely 'retrieve' from the queue or something. Since it'll just give you the list of drawables that are in the queue

guusw commented on shards/modules/gfx/feature.cpp:
```diff
@@ -146,6 +151,11 @@ struct FeatureShard {
   static SHTypesInfo inputTypes() { return CoreInfo::NoneType; }
   static SHTypesInfo outputTypes() { return ShardsTypes::Feature; }
 
+  static SHOptionalString help() { return SHCCSTR("This shard creates a feature object based on what was provided in the different parameters."); }
```
More straightforward to say "This shard creates a built-in or ready-made feature that performs a specific task" or something

guusw commented on shards/modules/gfx/feature.cpp:
```diff
@@ -178,7 +188,7 @@ struct FeatureShard {
   PARAM_PARAMVAR(
       _params, "Params",
-      "The parameters to expose to shaders, these default values can later be overriden by materials or drawable Params",
+      "The parameters to expose to shaders, these default values can later be modified.",
```
Maybe modified is a bit too vague, I wanted to clarify that this is where you add new parameters which can later be set per-object using something like:
```dart
GFX.Drawable(Params: {my-param: @f3(0.2 0.3 0.0)})
; or 
GFX.Material(Params: {my-param: @f3(0.2 0.3 0.0)}) = mat
GFX.Drawable(Material: mat)
```

guusw commented on shards/modules/gfx/gltf.cpp:
```diff
@@ -103,6 +103,19 @@ struct GLTFShard {
+  }
+
+  static SHOptionalString inputHelp() {
+    return SHCCSTR("This shard can take 2 different types of input. It can take a 4x4 transformation matrix containing the "
```
just `4x4 transformation matrix` is enough

guusw replied:
for compactness

guusw commented on shards/modules/gfx/steps.cpp:
```diff
@@ -264,17 +264,26 @@ struct DrawablePassShard {
+
   PARAM_EXT(ParamVar, _name, ShardsTypes::NameParameterInfo);
-  PARAM_PARAMVAR(_queue, "Queue", "The queue that this pass should render", {DrawQueueVarType});
+  PARAM_PARAMVAR(_queue, "Queue", "The drawables queue to use to create the render pass.", {DrawQueueVarType});
```
Maybe better to say: "The drawable queue to get drawables from" or "The drawable queue to draw objects from when rendering this pass"

ShahFragcolor committed:
Made changes requested

- changed bytes sequence to byte array
- made changes requested

ShahFragcolor committed:
Made requested changes

- changed Hash shards help text from "converts the input" to "outputs the hash of the input ..."
- Changed ToJson input help text to be more clear

ShahFragcolor committed:
Merge branch 'shah-docs-5' into shah-docs-8
